File: services/customers-service/src/main/java/br/com/deltaglobalbank/customers/infrastructure/persistence/repositories/JpaBankAccountRepository.java
package br.com.deltaglobalbank.customers.infrastructure.persistence.repositories;

import java.util.List;
import java.util.Set;
import java.util.UUID;

import br.com.deltaglobalbank.customers.infrastructure.persistence.entities.BankAccountEntity;
import org.springframework.data.jpa.repository.JpaRepository;

public interface JpaBankAccountRepository extends JpaRepository<BankAccountEntity, UUID> {
    List<BankAccountEntity> findAllByCustomerId(UUID customerId);
    List<BankAccountEntity> findAllByCustomerIdIn(Set<UUID> customerIds);
}

File: services/customers-service/src/main/kotlin/br/com/deltaglobalbank/customers/infrastructure/persistence/repositories/JpaBankAccountRepository.kt
package br.com.deltaglobalbank.customers.infrastructure.persistence.repositories

import br.com.deltaglobalbank.customers.infrastructure.persistence.entities.BankAccountEntity
import org.springframework.data.jpa.repository.JpaRepository
import java.util.UUID

interface JpaBankAccountRepository : JpaRepository<BankAccountEntity, UUID> {
    fun findAllByCustomerId(customerId: UUID): List<BankAccountEntity>
    fun findAllByCustomerIdIn(customerIds: Set<UUID>): List<BankAccountEntity>
}